# Purcell2020

This repository contains source code and supporting material for the manuscript:
Rectal swabs are a reliable method of assessing the colonic microbiome
Rachel Purcell et, al.

This manuscript is published at ...:
hyperlink

Sequencing data has been uploaded to SRA, the BioProject number is: PRJNA648177

Abstract:
Aims:
Advances in genome sequencing have enabled microbiome analysis in much greater detail than achievable with conventional culture methods; however, the ideal specimen type for sequencing is yet to be determined. Rectal swabs may offer a rapid and convenient modality for colonic microbiome analysis. The aim of this study is to evaluate the use of rectal swabs for microbiome analysis compared to faecal specimens.
Methods and Results:
Twenty health professionals participated in this study and provided a faecal specimen, a self-collected rectal swab and a rectal swab taken by a clinician. DNA was extracted and 16S rRNA sequencing was carried out for microbiome analysis.
Alpha diversity was higher in swabs compared to faecal specimens, however only for the clinician-obtained swabs was this difference significant.
Assessment of beta diversity consistently showed that few taxa were affected by the type of specimen analysed. Notably, there were very few taxa identified in clinician-obtained swabs that were not also found in the self-taken swabs.
Conclusions:
Rectal swabs are a reliable method of analysing the colonic microbiome.
Significance and Impact of Study:
Obtaining specimens for microbiome analysis is often time-critical due to the influence of therapy, such as antibiotics, on the microbiome. This study shows rectal swabs to be a convenient and reliable alternative to faecal specimens for microbiome analysis.


Notes:
