# Purcell2020

This repository contains source code and supporting material for the manuscript:  
"Rectal swabs are a reliable method of assessing the colonic microbiome", by Greg Turner,  Michael O'Grady, Daniel Hudson, Xochitl Morgan, Frank Frizelle, and Rachel Purcell. Int J Med Microbiol. 2022 Feb;312(2):151549.  doi: 10.1016/j.ijmm.2022.151549, PMID: 35114582
  
Sequencing data has been uploaded to SRA, the BioProject number is: PRJNA648177  
  
Abstract:
Background: Advances in genome sequencing have enabled detailed microbiome analysis; however, the ideal specimen type for sequencing is yet to be determined. Rectal swabs may offer a rapid and convenient modality for colonic microbiome analysis. The aim of this study is to evaluate the use of rectal swabs compared to faecal specimens.

Methods and results: Twenty health professionals participated in this study and provided a faecal specimen, a self-collected rectal swab and a rectal swab taken by a clinician. DNA was extracted and 16S rRNA gene sequencing was carried out for microbiome analysis. Alpha diversity was higher in swabs compared to faecal specimens; however, the difference was only significant when comparing clinician-obtained swabs to faeces. Analysis of beta diversity consistently showed that few taxa were affected by sample type. We found sample type accounted for only 6.8% of community variation (R2 = 0.067, p < 0.001, permanova). Notably, there were only six genera identified in clinician-obtained swabs that were not also found in the self-taken swabs.

Conclusions: Both self-collected and clinician obtained rectal swabs are a reliable method of analysing the colonic microbiome. Obtaining specimens for microbiome analysis is often time-critical due to therapy, such as antibiotics, influencing the microbiome. Rectal swabs are shown to be a valid and convenient modality for microbiome analysis. 


